### PR TITLE
NEXUS-22735: Conan search command improvement

### PR DESCRIPTION
**Pull request by mlukaretkyi**
https://issues.sonatype.org/browse/NEXUS-22735

### CODE REVIEW

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyControllerV1.java [hunk 1]
```diff
@@ -16,19 +16,24 @@
...
 {
   @Inject
-  ProxyHandler proxyHandler;
+  private ConanProxyHandlers conanProxyHandlers;
 
   public void attach(final Router.Builder builder) {
-    createGetRoutes(builder, proxyHandler, proxyHandler, proxyHandler, proxyHandler);
+    // query implementation
+    createRoute(builder, ConanRoutes.searchQuery(), null, conanProxyHandlers.searchQueryHandler);
+    // query implementation
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyControllerV1.java]: Double comment?

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java [hunk 2]
```diff
@@ -286,19 +287,43 @@ protected void indicateVerified(final Context context, final Content content, fi
...
 
+  @TransactionalTouchBlob
+  @Nullable
+  protected String getUrlForConanManifest(final ConanCoords coords) {
+    String downloadUrlsAssetPath = ConanProxyHelper.getProxyAssetPath(coords, DOWNLOAD_URL);
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Asset downloadUrlAsset = findAsset(tx, tx.findBucket(getRepository()), downloadUrlsAssetPath);
+    if (downloadUrlAsset == null) {
+      // TODO Looks like it was search request. So let's look up conanmanifest url from digest
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java]: Add ticket number to TODO

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyHelper.groovy [hunk 3]
```diff
@@ -40,15 +40,19 @@ import static org.sonatype.repository.conan.internal.metadata.ConanMetadata.VERS
  */
 class ConanProxyHelper
 {
-  public static final List<AssetKind> DOWNLOAD_ASSET_KINDS = ImmutableList.of(AssetKind.DOWNLOAD_URL, AssetKind.DIGEST, AssetKind.CONAN_PACKAGE_SNAPSHOT)
+  public static final List<AssetKind> METADATA_ASSET_KINDS = ImmutableList.of(
+      AssetKind.DOWNLOAD_URL,
+      AssetKind.DIGEST,
+      AssetKind.CONAN_PACKAGE_SNAPSHOT
+  )
```
Comment (igorudovika) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyHelper.groovy]: I do not agree with this change. Conanmanifest or Conaninfo is also metadata therefore we are adding more confuse here. Mayby URL_ASSET_KIND(not sure is it true for CONAN_PACKAGE_SNAPSHOT) or DOWNLOAD_METADATA_ASSET_KINDS. Or leave it as it.

Comment (mlukaretkyi) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyHelper.groovy]: I returned back prev name
https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/98/commits/3c7d54608b90e0730d43d62d549032b70bb4d380

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java (hunk 2 above)
Comment (mlukaretkyi) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java]: fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/98/commits/3c7d54608b90e0730d43d62d549032b70bb4d380

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyControllerV1.java (hunk 1 above)
Comment (mlukaretkyi) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyControllerV1.java]: removed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/98/commits/3c7d54608b90e0730d43d62d549032b70bb4d380

**Pull request merged**